internal/controllers/machinedeployment/mdutil/util_test.go

```diff
@@ -755,6 +765,153 @@ func TestAnnotationUtils(t *testing.T) {
+			ms:         nil,
+			want: map[string]string{
+				"key1":                              "value1",
+				clusterv1.RevisionAnnotation:        "2",
+				clusterv1.DesiredReplicasAnnotation: "3",
+				clusterv1.MaxReplicasAnnotation:     "4",
+			},
+			wantErr: false,
+		},
+		{
+			name:       "Calculating annotations for a existing MachineSet",
+			deployment: &deployment,
+			oldMSs:     nil,
+			ms:         machineSetWithRevisionAndHistory("1", ""),
+			want: map[string]string{
+				"key1":                              "value1",
+				clusterv1.RevisionAnnotation:        "1",
+				clusterv1.DesiredReplicasAnnotation: "3",
+				clusterv1.MaxReplicasAnnotation:     "4",
+			},
+			wantErr: false,
+		},
+		{
+			name:       "Calculating annotations for a existing MachineSet - old MSs exist",
+			deployment: &deployment,
+			oldMSs: []*clusterv1.MachineSet{
+				machineSetWithRevisionAndHistory("1", ""),
+				machineSetWithRevisionAndHistory("2", ""),
+			},
+			ms: machineSetWithRevisionAndHistory("1", ""),
+			want: map[string]string{
+				"key1":                              "value1",
+				clusterv1.RevisionAnnotation:        "3",
+				clusterv1.RevisionHistoryAnnotation: "1",
+				clusterv1.DesiredReplicasAnnotation: "3",
+				clusterv1.MaxReplicasAnnotation:     "4",
+			},
+			wantErr: false,
+		},
+		{
+			name:       "Calculating annotations for a existing MachineSet - old MSs exist - existing revision is greater",
+			deployment: &deployment,
+			oldMSs: []*clusterv1.MachineSet{
+				machineSetWithRevisionAndHistory("1", ""),
+				machineSetWithRevisionAndHistory("2", ""),
+			},
+			ms: machineSetWithRevisionAndHistory("4", ""),
+			want: map[string]string{
+				"key1":                              "value1",
+				clusterv1.RevisionAnnotation:        "4",
+				clusterv1.DesiredReplicasAnnotation: "3",
+				clusterv1.MaxReplicasAnnotation:     "4",
+			},
+			wantErr: false,
+		},
+		{
+			name:       "Calculating annotations for a existing MachineSet - old MSs exist - ms already has revision history",
+			deployment: &deployment,
+			oldMSs: []*clusterv1.MachineSet{
+				machineSetWithRevisionAndHistory("3", ""),
+				machineSetWithRevisionAndHistory("4", ""),
+			},
+			ms: machineSetWithRevisionAndHistory("2", "1"),
+			want: map[string]string{
+				"key1":                              "value1",
+				clusterv1.RevisionAnnotation:        "5",
+				clusterv1.RevisionHistoryAnnotation: "1,2",
+				clusterv1.DesiredReplicasAnnotation: "3",
+				clusterv1.MaxReplicasAnnotation:     "4",
+			},
+			wantErr: false,
+		},
+	}
+
+	log := klogr.New()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			got, err := ComputeMachineSetAnnotations(log, tt.deployment, tt.oldMSs, tt.ms)
+			if tt.wantErr {
+				g.Expect(err).ShouldNot(BeNil())
+			} else {
+				g.Expect(err).Should(BeNil())
+				for k, v := range tt.want {
+					g.Expect(got).Should(HaveKeyWithValue(k, v))
+				}
```

Comment:
```suggestion
				g.Expect(got).Should(Equal(tt.want))
```
Otherwise we wouldn't detect additional annotations in got